packages/beacon-node/src/chain/validation/blobSidecar.ts

```diff
@@ -0,0 +1,154 @@
+import bls from "@chainsafe/bls";
+import {CoordType} from "@chainsafe/bls/types";
+import {ChainForkConfig} from "@lodestar/config";
+import {deneb, Root, Slot, bellatrix} from "@lodestar/types";
+import {bytesToBigInt, toHex} from "@lodestar/utils";
+import {BYTES_PER_FIELD_ELEMENT, FIELD_ELEMENTS_PER_BLOB} from "@lodestar/params";
+import {verifyKzgCommitmentsAgainstTransactions} from "@lodestar/state-transition";
+import {BlobSidecarError, BlobSidecarErrorCode} from "../errors/blobSidecarError.js";
+import {GossipAction} from "../errors/gossipValidation.js";
+import {byteArrayEquals} from "../../util/bytes.js";
+import {ckzg} from "../../util/kzg.js";
+// import {IBeaconChain} from "../interface.js";
+
+const BLS_MODULUS = BigInt("52435875175126190479447740508185965837690552500527637822603658699938581184513");
+
+export function validateGossipBlobSidecar(
+  // TODO: freetheblobs use config and chain for some validations
+  _config: ChainForkConfig,
+  blobSidecar: deneb.BlobSidecar,
+  gossipIndex: number
+): void {
+  // TODO: freetheblobs - validate blob signature
+  if (blobSidecar.index !== gossipIndex) {
+    throw new BlobSidecarError(GossipAction.REJECT, {
+      code: BlobSidecarErrorCode.INVALID_INDEX,
+      blobIdx: blobSidecar.index,
+      gossipIndex,
+    });
+  }
+  // [REJECT] the sidecar.blobs are all well formatted, i.e. the BLSFieldElement in valid range (x < BLS_MODULUS).
+  if (!blobIsValidRange(blobSidecar.blob)) {
+    throw new BlobSidecarError(GossipAction.REJECT, {
+      code: BlobSidecarErrorCode.INVALID_BLOB,
+      blobIdx: blobSidecar.index,
+    });
+  }
+
+  if (!blsKeyValidate(blobSidecar.kzgCommitment)) {
+    throw new BlobSidecarError(GossipAction.REJECT, {
+      code: BlobSidecarErrorCode.INVALID_KZG,
+      blobIdx: blobSidecar.index,
+    });
+  }
+
+  // verifyKzgCommitmentsAgainstTransactions will need to be done while block verification when block
+  // would also be available
+
+  // [REJECT] The KZG proof is a correctly encoded compressed BLS G1 Point
+  // -- i.e. blsKeyValidate(blobs_sidecar.kzg_aggregated_proof)
+  if (!blsKeyValidate(blobSidecar.kzgProof)) {
+    throw new BlobSidecarError(GossipAction.REJECT, {
+      code: BlobSidecarErrorCode.INVALID_KZG_PROOF,
+      blobIdx: blobSidecar.index,
+    });
+  }
+
+  validateBlobsAndProofs([blobSidecar.kzgCommitment], [blobSidecar.blob], [blobSidecar.kzgProof]);
```

Comment:
@g11tech educated me on why we have BlsKeyValidate and BlobIsValidRange here and we could remove those as they are being done when we validate the proof with the blob.

(Whether you will be able to get the same amount of error information will depend on c-kzg errors)

Reply:
removed explicit validations to totally rely on underlying kzg lib